src/App.tsx

```diff
@@ -1,24 +1,193 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { SortBy } from './types/SortBy';
+import { addTodo, deleteTodo } from './api/todos';
+import { Todos } from './components/Todos';
+import { useGetTodos } from './hooks';
 
-const USER_ID = 0;
+const USER_ID = 11361;
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.all);
+  const [selectedTodo, setSelectedTodo] = useState<number[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoadingTodo, setIsLoadingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const {
+    isLoading,
+    todos,
+    errorMessage,
+    todosNotCompleted,
+    handleError,
+    handleIsTodoDeleted,
+  } = useGetTodos(USER_ID, tempTodo);
+
+  const deleteOneTodo = async (todoId: number) => {
+    try {
+      await deleteTodo(USER_ID, todoId);
+    } catch (error) {
+      handleError('Unable to delete a todo');
+    }
+
+    handleIsTodoDeleted(true);
+    setSelectedTodo(prevSelectedTodo => [...prevSelectedTodo, todoId]);
+  };
+
+  const handleDeleteTodo = (value: number) => deleteOneTodo(value);
+
+  const handleKeyDown = async (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (!inputValue) {
+        handleError("Title can't be empty");
+      }
+
+      setIsLoadingTodo(true);
+
+      setTempTodo({
```

Comment:
The use of this `setTempTodo` seems you probably want to make some optimistic changes on the Todo list, but it could cause strange behavior on the UI because on the `finally` statement the state is set to null.
Also the state `tempTodo` is used as a parameter in the `useGetTodos` hook and this value is changed twice at least and for every change the `useGetTodos` hook is executed again.

Reply:
It works like requiments said: 
immediately after sending a request create a todo with id: 0 and save it to the tempTodo variable in the state (NOT to the todos array);
show an independent TodoItem after the list if tempTodo is not null;
temp TodoItem should have the loader (check the original markup);
tempTodo should to set as a new todo and then set to null when finally todo is added to todods, so it causes thath useGetTodos is executed twice